src/bin/jit.rs

```diff
@@ -0,0 +1,397 @@
+/*fn main() {
+    let module = rcc::initialize_jit_module();
+    let (jit, warnings) = compile(module, buf, &opt);
+    handle_warnings(warnings, file_id, file_db);
+    let mut jit = RccJIT::from_module(jit?);
+    jit.finalize();
+    let main_function = jit.get_compiled_function("main");
+    if let Some(main_function) = main_function {
+        let args = std::env::args();
+        let argc = args.len() as i32;
+        let vec_args = args
+            .map(|string| std::ffi::CString::new(string).unwrap())
+            .collect::<Vec<std::ffi::CString>>();
+        let pointer = vec_args
+            .iter()
+            .map(|cstr| cstr.as_ptr() as *const u8)
+            .collect::<Vec<*const u8>>()
+            .as_ptr() as *const *const u8;
+        let main: unsafe extern "C" fn(i32, *const *const u8) -> i32 =
+            unsafe { std::mem::transmute(main_function) };
+        let _ = unsafe { main(argc, pointer) };
+    }
+    Ok(())
+}
+*/
+
+use std::collections::VecDeque;
+use std::fs::File;
+use std::io::{self, Read};
+use std::num::NonZeroUsize;
+use std::path::PathBuf;
+use std::process;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+extern crate ansi_term;
+extern crate codespan;
+#[cfg(debug_assertions)]
+extern crate color_backtrace;
+extern crate env_logger;
+extern crate log;
+extern crate pico_args;
+extern crate rcc;
+
+use ansi_term::{ANSIString, Colour};
+use codespan::{FileId, Files};
+use pico_args::Arguments;
+use rcc::{
+    compile,
+    data::{
+        error::{CompileWarning, RecoverableResult},
+        lex::Location,
+    },
+    preprocess, utils, Error, Opt, RccJIT,
+};
+use std::ffi::OsStr;
+
+static ERRORS: AtomicUsize = AtomicUsize::new(0);
+static WARNINGS: AtomicUsize = AtomicUsize::new(0);
+
+const HELP: &str = concat!(
```

Comment:
Hmm ... I know I told you to split the binaries up, but seeing all the duplicate code makes me unhappy... Maybe switch it back after all? I don't know what I want 😆


Certainly most of this can be reused, maybe you could make a bin_utils module or similar? I'm open to suggestions.